utilities/vmware_guest_operations.py

```
@@ -0,0 +1,249 @@
+            vm=vm,
+            auth=auth,
+            command=script,
+            vcenter_host=vcenter_host,
+        )
+    except GuestCommandError:
+        LOGGER.warning(f"Guest command failed for VM {vm.name}, IP origin detection incomplete")
+        raise
+
+    origins = _parse_nmcli_ip_origins(output)
+    if not origins:
+        LOGGER.warning(f"No IP origin data detected via Guest Operations for VM {vm.name}")
```

Ruff (0.15.5)

[warning] Logging statement uses f-string

(G004)